### PR TITLE
inclusive language: use "controller" instead of "master"

### PR DESCRIPTION
**Pull request by ffmancera**
In order to promote a diverse and inclusive environment we are replacing
all the reference to "master" for "controller".

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 linux-system-roles/network
 ==========================
 [![Coverage Status](https://coveralls.io/repos/github/linux-system-roles/network/badge.svg)](https://coveralls.io/github/linux-system-roles/network)
-[![Travis Build Status](https://travis-ci.com/linux-system-roles/network.svg?branch=master)](https://travis-ci.com/linux-system-roles/network)
+[![Travis Build Status](https://travis-ci.com/linux-system-roles/network.svg?branch=main)](https://travis-ci.com/linux-system-roles/network)
```
Comment (tyll) [inline README.md:4]: good catch. It does not belong in this PR, strictly speaking.

File: README.md [hunk 2]
```diff
@@ -822,4 +822,4 @@ it is not clear that rollback will result in a working configuration.
 
 
 *Want to contribute? Take a look at our [contributing
-guidelines](https://github.com/linux-system-roles/network/blob/master/contributing.md)!*
```
Comment (tyll) [inline README.md:825]: Please group this together with other changes about the branch name. Thanks.

File: tox.ini [hunk 3]
```diff
@@ -170,7 +170,7 @@ deps =
     ansible
     jmespath
 commands =
-    bash {toxinidir}/.travis/runcollection.sh {toxworkdir} master 'black,flake8,yamllint,py38'
+    bash {toxinidir}/.travis/runcollection.sh {toxworkdir} main 'black,flake8,yamllint,py38'
```
Comment (tyll) [inline tox.ini:173]: This needs to be fixed in the CI template.

**Comment by tyll** [comment]
Please add tests that show that the deprecation actually works.

File: tox.ini (hunk 3 above)
Comment (richm) [reply tox.ini:173]: @nhosoi ^^^
* need to make the runcollection.sh script detect if the default main branch is `main` or `master` - maybe use `main` if it exists, otherwise, use `master` (if `STABLE_TAG` is unset)
* need to merge the shellcheck PR into network so that we can use the default envlist rather than having a custom envlist here to workaround the missing shellcheck PR . . .

Comment (nhosoi) [reply tox.ini:173]: Correction: the branch name `master` is for downloading lsr_role2collection.py. So, there is no need to replace it with main. I'm just working on the second item from Rich.

Comment (richm) [reply tox.ini:173]: hmm - well if the network role is trying to remove every trace of exclusive language, then the use of `master` here might be problematic . . . but this particular problem goes away once we have tox-lsr

Comment (nhosoi) [reply tox.ini:173]: Then, can we leave it for now?

Comment (nhosoi) [reply tox.ini:173]: I've opened a [pr](https://github.com/linux-system-roles/network/pull/307) to sync with the template, which eliminates the master option.

Please note that the default branch name for the lsr_role2collection.py script in .travis/runcollection.sh remains. I hope it's ok.
https://github.com/linux-system-roles/network/pull/307/files#diff-91dffa6ccef3dee4f1536129fe5b9d5b9c87eb357ba4f241b6d1daa6f230a3adR16

Comment (tyll) [reply tox.ini:173]: Would it be possible to change the default branch name for the script to main?

**Comment by ffmancera** [comment]
> Please add tests that show that the deprecation actually works.

Done! I've added a new test case.

File: tox.ini (hunk 3 above)
Comment (ffmancera) [reply tox.ini]: Thank you! I've removed my changes. In addition it would be very nice to rename all "master" branchs to "main".

File: README.md (hunk 2 above)
Comment (ffmancera) [reply README.md]: Moved to another PR.

File: README.md (hunk 1 above)
Comment (ffmancera) [reply README.md]: Moved to another PR. Thanks!

**Comment by tyll** [comment]
Please check the Travis CI results

**Comment by ffmancera** [comment]
> Please check the Travis CI results

It was a formatting error. Fixed!

**Comment by ffmancera** [comment]
I plan to use the test file introduced in this PR on #303

File: tests/playbooks/tests_bond_deprecated.yml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        msg: "##################################################"
+    - import_tasks: tasks/create_test_interfaces_with_dhcp.yml
+    - import_tasks: tasks/assert_device_present.yml
+      vars:
+        interface: "{{ dhcp_interface1 }}"
+    - import_tasks: tasks/assert_device_present.yml
+      vars:
+        interface: "{{ dhcp_interface2 }}"
+    - block:
+        - name: "TEST Add Bond with 2 slaves using deprecated master"
```
Comment (tyll) [inline tests/playbooks/tests_bond_deprecated.yml:23]: IMHO it is not clear from the name what is deprecated/tested here.

**Comment by tyll** [comment]
> > Please add tests that show that the deprecation actually works.
> 
> Done! I've added a new test case.

I did not find any unit tests, did you add them? The integration test only tests one scenario. Please consider testing all relevant scenarios (compare the tests for previously deprecated items).

**Comment by ffmancera** [comment]
> > > Please add tests that show that the deprecation actually works.
> > 
> > 
> > Done! I've added a new test case.
> 
> I did not find any unit tests, did you add them? The integration test only tests one scenario. Please consider testing all relevant scenarios (compare the tests for previously deprecated items).

I've created a unit test for this. IMO, that should be enough, the test is checking the "master" argument is being translated into "controller".

File: tests/playbooks/tests_bond_deprecated.yml (hunk 4 above)
Comment (ffmancera) [reply tests/playbooks/tests_bond_deprecated.yml]: I've clarified it. Thanks!

File: tests/unit/test_network_connections.py [hunk 5]
```diff
@@ -2998,6 +2998,106 @@ def test_valid_persistent_state(self):
...
+        input_connections = [
+            {
+                "name": "prod2",
+                "state": "up",
+                "type": "bridge",
+                "interface_name": "bridge2",
+                "ip": {"dhcp4": False, "auto6": False},
+            },
+            {
+                "name": "prod2-slave1",
```
Comment (tyll) [inline tests/unit/test_network_connections.py:3014]: the name can use inclusive language, too.

File: tests/unit/test_network_connections.py [hunk 6]
```diff
@@ -2998,6 +2998,106 @@ def test_valid_persistent_state(self):
...
+    def test_set_deprecated_master(self):
+        """
+        When passing the deprecated "master" it is updated to "controller".
+        """
+        input_connections = [
+            {
+                "name": "prod2",
+                "state": "up",
+                "type": "bridge",
+                "interface_name": "bridge2",
```
Comment (tyll) [inline tests/unit/test_network_connections.py:3010]: interface_name can be omitted, can't it?

File: tests/unit/test_network_connections.py [hunk 7]
```diff
@@ -2998,6 +2998,106 @@ def test_valid_persistent_state(self):
...
+            },
+            {
+                "name": "prod2-slave1",
+                "state": "up",
+                "type": "ethernet",
+                "interface_name": "eth1",
+                "master": "prod2",
+            },
+        ]
+        expected_connections = [
```
Comment (tyll) [inline tests/unit/test_network_connections.py:3021]: This big list of expected connections. Would it be possible to make it as concise as the other tests around deprecation like https://github.com/linux-system-roles/network/blob/main/tests/unit/test_network_connections.py#L2951 ?

In this case it would be nice to only:
1. Assert that controller: "prod2" is in the output of the validation
2. Assert that master is not included in the output of the validation.

**Review by tyll — COMMENTED** [review]
> > > > Please add tests that show that the deprecation actually works.
> > > 
> > > 
> > > Done! I've added a new test case.
> > 
> > 
> > I did not find any unit tests, did you add them? The integration test only tests one scenario. Please consider testing all relevant scenarios (compare the tests for previously deprecated items).
> 
> I've created a unit test for this. IMO, that should be enough, the test is checking the "master" argument is being translated into "controller".

What about setting both `master` and `controller`, what is expected to happen there? IMHO the expected behavior for this should be tested as well. Will there be a difference if master/controller specify different profiles?

File: tests/unit/test_network_connections.py (hunk 5 above)
Comment (ffmancera) [reply tests/unit/test_network_connections.py:3014]: Oops. It should, I just copied it from another example. The example will use inclusive language on #303, I will fix the new one.

File: tests/unit/test_network_connections.py (hunk 6 above)
Comment (ffmancera) [reply tests/unit/test_network_connections.py:3010]: I think yes, let me try.

File: tests/unit/test_network_connections.py (hunk 7 above)
Comment (ffmancera) [reply tests/unit/test_network_connections.py:3021]: I will try that. Thanks!

**Comment by cathay4t** [comment]
@ffmancera Please let me know after you fixed all the review feedback and new patch is ready for review. Thank you!

File: tests/unit/test_network_connections.py (hunk 6 above)
Comment (ffmancera) [reply tests/unit/test_network_connections.py:3010]: Fixed!

File: tests/unit/test_network_connections.py (hunk 7 above)
Comment (ffmancera) [reply tests/unit/test_network_connections.py:3021]: Fixed, I am just checking controller is present in both connections.

**Comment by ffmancera** [comment]
@cathay4t Hi! This is ready for review.
